packages/js/data/src/onboarding/with-onboarding-hydration.tsx

```diff
@@ -21,39 +20,54 @@ export const withOnboardingHydration = ( data: {
 		( OriginalComponent ) => ( props ) => {
 			const onboardingRef = useRef( data );
```

Comment:
This ref probably never had to be there? The `data` value is constant, it's passed once for all when creating a HOC with `withOnboardingHydration( data )`. There's no need for each wrapped component to store its own copy in a local ref. The code can just reference the `data` variable in the closure. Just like it references `hydratedProfileItems`, or `group` (in other usages of this pattern).